### ISSUE TITLE
Modify desktop cursor sizing and styling for additional visual contrast and uniformity

### ISSUE THREAD

**Issue by shiftyscales** [Yellow-Dog-Man/Resonite-Issues#3124]
### Is your feature request related to a problem? Please describe.

The current cursor set in Resonite has a lot of little inconsistencies and issues in its presentation that can make the cursor difficult to see due to low contrast, and small sizing.

In specific the default cursor (not highlighting any interactive elements) is an extremely low contrast gray that is very small and easy to get lost.

E.g. here's the default cursor hovering over a gizmo.

![Image](https://github.com/user-attachments/assets/cc3c9f92-1df6-4494-8c0b-c0321e64a120)

Cursor styles are also inconsistent. The default is a gray cross that is much smaller than other cursors.

---

The grabbable cursor isn't filled with colour like the other cursor types, but is slightly larger.

![Image](https://github.com/user-attachments/assets/cbad0096-1747-4ab4-96e9-29adb52dd62a)

---

The interaction cursor is filled with colour, but is also much smaller than the grabbable cursor.

![Image](https://github.com/user-attachments/assets/ed483b88-5067-4471-9ed2-43a41dcd24ca)

On the menu we use a modified "interaction" cursor that is purple instead of blue- it has the same issues with size and contrast- particularly when it is hovered over various parts of the interface.

![Image](https://github.com/user-attachments/assets/5fc5b9cf-7251-472a-9354-ae7526b0b81a)

---
Crops from the above images showcasing the low contrast and small sizing of the existing cursor set.

![Image](https://github.com/user-attachments/assets/039ca742-5f83-420d-be94-eaef00415668)
![Image](https://github.com/user-attachments/assets/df6f5d0d-45a0-4f86-9485-c0549c0de50e)
![Image](https://github.com/user-attachments/assets/05c35392-8eb3-4528-ba87-cbfea8af53cd)
![Image](https://github.com/user-attachments/assets/186d38b8-30c4-4951-a8f8-b0c27814f282)

The cursors do appear to have a hint of a black outline which is good- but the cursor is not displayed at a large enough size for that to provide proper contrast- the outline of the cursors are rendered so small they blend in to the colour of the cursor itself rather than properly outlining it.

### Describe the solution you'd like

Ensuring that the cursors have a consistent ~2px black outline and sizing at the least would help with contrast- but we may wish to possibly consider working with cursor colours outside of our brand palette too- keeping the cursor branded / coloured like the rest of Resonite means that it blends in with official content which uses the same colours.

If we used a neutral / bright white cursor with a dark outline- it would work well in a majority of cases. We can still modify the cursor contextually, but this is a place where it's probably better that we keep things simple as a default.

### Describe alternatives you've considered

For desktop purposes- we could possibly even just pass through the Windows cursor / provide an option to not modify the cursor at all- in this way the user can make use of any modified / accessible cursor they are already using on the rest of their system since Windows already has built in accessibility settings to scale / ensure the cursor has high contrast as-needed.

### Additional Context

Creating this issue following up from some discussion on an internal issue where this topic came up- in specific in the context of cursor customization. This issue is being made to focus on enhancing the default cursor to be more visible / accessible as a default. Modification of the cursor would be better handled as separate requests, e.g. #1356 / #2902.

### Requesters

@shiftyscales @ProbablePrime @Frooxius

**Comment by ProbablePrime** [comment]
Somehow forgot to keep this one up to date.

**Comment by ProbablePrime** [comment]
The sizing is probably all I'll be able to tackle for a first "batch", but Froox and I were talking about custom Cursor images.

An outline is tricky, the shaders we're using in the area cannot draw outlines, but I may be able to add a "brightness" modifier to the cursors and lasers quite easily.

**Comment by ProbablePrime** [comment]
You can see the latest changes here: https://www.youtube.com/watch?v=a8qQ9luZNNM

I'm happy with these results enough to pause for now and get these changes in. But I do want to do more later.

For later(subsequent initiatives) some ideas:
1. Disable Resonite cursors all up - This will show the windows cursors if this setting is enabled. This might be good for some users for accesibility. I want to see how this behaves with our lasers, but for Desktop mode, I'm confident that this one is simple enough
2. Outlines - Shifty mentioned that an outline might help to see the cursors for some people. I want to do this in a way that the color can be changed too.
3. Custom cursor images - We eventually want to make settings able to interact with images, which will allow for custom cursors
4. Cursor/Interaction Components to control cursors - Internally we use an "interactiondescription" structure for cursors. We could update this to allow more control in world, which could allow for certain objects to have differing cursors. I still need to play with the design of this one more to see if its a good idea.

**Comment by Frooxius** [comment]
Merged the overhaul in 2024.12.16.1304! Thank you!
